From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 May 2019 19:49:19 +0200
Subject: [PATCH] {MAYBE UPS} drivers/misc/habanalabs: fix cast on 32-bit
 compilation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc for x86-32 reports:

    drivers/misc/habanalabs/habanalabs_ioctl.c: In function ‘debug_coresight’:
    drivers/misc/habanalabs/habanalabs_ioctl.c:143:23: error: cast to
    pointer from integer of different size
    [-Werror=int-to-pointer-cast]
       input = memdup_user((const void __user *) args->input_ptr,
                           ^
---
 drivers/misc/habanalabs/habanalabs_ioctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/habanalabs/habanalabs_ioctl.c b/drivers/misc/habanalabs/habanalabs_ioctl.c
index eeefb22023e9..905a49973ea2 100644
--- a/drivers/misc/habanalabs/habanalabs_ioctl.c
+++ b/drivers/misc/habanalabs/habanalabs_ioctl.c
@@ -140,7 +140,7 @@ static int debug_coresight(struct hl_device *hdev, struct hl_debug_args *args)
 	params->op = args->op;
 
 	if (args->input_ptr && args->input_size) {
-		input = memdup_user((const void __user *) args->input_ptr,
+		input = memdup_user((const void __user *)(unsigned long) args->input_ptr,
 					args->input_size);
 		if (IS_ERR(input)) {
 			rc = PTR_ERR(input);
-- 
